pom.xml

```diff
@@ -188,15 +188,13 @@
         </plugin>
         <plugin>
           <artifactId>maven-surefire-plugin</artifactId>
-          <version>2.7.2</version>
+          <version>2.22.2</version>
           <configuration>
             <includes>
               <include>${test.include}</include>
             </includes>
-            <!-- By having our own entries here, we also override the default exclusion filter, which excludes all nested classes. -->
-            <excludes>
-            </excludes>
-            <redirectTestOutputToFile>true</redirectTestOutputToFile>
+            <redirectTestOutputToFile>false</redirectTestOutputToFile>
+            <forkCount>0</forkCount>
```

Comment:
@vikram-gaur This was the magic to make it work.

The forking really didn't play nice with whatever containerizing github actions are doing, so by disabling forks it all runs in the main process.

This does mean that a really bad failure could kill the whole maven process, but I think that's fine. We should get a stack trace in the unlikely case that happens.